pkg/ipamd/ipamd.go

```diff
@@ -743,26 +743,30 @@ func (c *IPAMContext) setupENI(eni string, eniMetadata awsutils.ENIMetadata, tru
 	if err != nil && err.Error() != datastore.DuplicatedENIError {
 		return errors.Wrapf(err, "failed to add ENI %s to data store", eni)
 	}
+	// Store the primary IP of the ENI
+	c.primaryIP[eni] = eniMetadata.PrimaryIPv4Address()
```

Comment:
Are you sure there won't be any concurrent access for this (given that your not using locks while accessing the map)?

Reply:
Yes, `c.primaryIP` is only accessed in `ipamd.go`, that is only ever running in a single go routine. That said, we should remove the ENI from the map if we remove the ENI from the datastore....